storage/object_storage_client.go

```diff
@@ -0,0 +1,83 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"os"
+)
+
+// References:
+// - S3 client:        https://docs.aws.amazon.com/code-library/latest/ug/go_2_s3_code_examples.html
+// - Transfer Manager: https://aws.github.io/aws-sdk-go-v2/docs/sdk-utilities/s3/
+
+type BucketBasics struct {
+	S3Client *s3.Client
+}
+
+const (
+	uploadPartSize   = 5 * 1024 * 1024 // 5 MiB
+	downloadPartSize = 5 * 1024 * 1024 // 5 MiB
```

Comment:
It seems 8MiB or 16MiB is a sweet spot according to https://www.vldb.org/pvldb/vol16/p2769-durner.pdf . But I'm not sure.

Reply:
fixed